### PR TITLE
Priority

### PR DESCRIPTION
**Pull request by ptrompeter**

### CODE REVIEW

File: src/priority.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class Priority(object):
+    """Return."""
+
+    def __init__(self, values=[]):
+        """Return."""
+        self.plist = []
+        for i in values:
+            self.push(i)
```
Comment (joelstanner) [inline src/priority.py]: Nice. Many students forgot to implement this functionality.

File: src/priority.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        popped_value = self.plist[0]
+        startval = self.plist.pop()
+        pos = 0
+        self.plist[pos] = startval
+        while pos < len(self.plist) // 2:
+            priority = self._higher_priority_child(pos)
+            if self.plist[priority] < self.plist[pos]:
+                self.plist[pos], self.plist[priority] = self.plist[priority], self.plist[pos]
+                pos = priority
+            else: break
```
Comment (joelstanner) [inline src/priority.py]: Avoid multiple statements on one line.

File: tests/test_priority.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+def test_pri_push2():
+    test_pri = Priority()
+    test_pri.push((3, 'a'))
+    test_pri.push((3, 'b'))
+    test_pri.push((3, 'c'))
+    test_pri.push((3, 'd'))
+    test_pri.push((2, 'a'))
+    assert test_pri.plist[0] == (2, 'a')
+
```
Comment (joelstanner) [inline tests/test_priority.py]: Be consistent with blank lines between functions. You have some with 1 line, and some with 2. The flake8 standard we are using for the class is 2 lines.

File: tests/test_priority.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    assert test_pri.plist[0][0] == 1
+
+def test_pri_pop3():
+    test_pri = Priority(TEST_LIST)
+    test_pri.pop()
+    test_pri.pop()
+    test_pri.pop()
+    assert test_pri.plist[0][0] == 2
+
+
```
Comment (joelstanner) [inline tests/test_priority.py]: Make sure your linter is catching excessive whitespace

Comment (joelstanner) [inline tests/test_priority.py]: Write tests that cover all of the methods of your data structure. You are missing tests for `peek()`, for example. Add edge case tests. What happens when you pop from a Priority Queue with only 1 item?

File: src/priority.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+class Priority(object):
+    """Return."""
```
Comment (joelstanner) [inline src/priority.py]: When you are composing these data structures, do your best to use docstrings to showcase your knowledge of how the structure works. Also use the readme to compose a general overview of all of your data structures. When you have docstrings like 'Return.', this is what your class looks like when someone asks for documentation on something like iPython:

```python
In [74]: pq?
Type:           Priority
String form:    <priority.Priority object at 0x10ced3eb8>
File:           ~/Documents/code_fellows/401python-d2/sandbox/ptrompeter/data-structures/src/priority.py
Docstring:      Return.
Init docstring: Return.
```

It would be best to just have no docstring as opposed to this one.
